tuf/data/roles.go

```diff
@@ -69,6 +69,16 @@ func ValidRole(name string) bool {
 	return false
 }
 
+// IsBaseRole checks if a role name is valid base role
+func IsBaseRole(name string) bool {
+	for _, v := range BaseRoles {
+		if name == v {
+			return true
+		}
+	}
+	return false
+}
+
 // IsDelegation checks if the role is a delegation or a root role
 func IsDelegation(role string) bool {
```

Comment:
Maybe a rename is more fitting?  I agree that we'll use type checking, but this method shouldn't be used check a role itself.  I'm think we should keep it for input validation for delegation names before we create or operate on typed roles, like on notary CLI.